src/types/JwtVerifier.ts

```diff
@@ -0,0 +1,91 @@
+import { calculateJwkThumbprintUri, getDigestAlgorithmFromJwkThumbprintUri } from '../helpers';
+
+import SIOPErrors from './Errors';
+import { JWK, JwtHeader, JwtPayload } from './JWT.types';
+
+export type JwtVerificationType = 'id-token' | 'request-object';
+
+interface DidJwtVerifier {
+  method: 'did';
+  type: JwtVerificationType;
+
+  didUrl: string;
+}
```

Comment:
For type inference it may be easier to have  base type where they all extend from:



```
interface JwtVerifierBase {
 method: string
 jwtType: JwtVerificationType
}

Also maybe rename type to jwtType?